File: algorithms/1248. Count Number of Nice Subarrays/solution.py
class Solution(object):
    def numberOfSubarrays(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        cache, total, result = collections.defaultdict(int), 0, 0
        cache[0] = 1
        for n in nums:
            total += n & 1
            cache[total] += 1
            result += cache[total - k]
        return result

File: algorithms/1248. Count Number of Nice Subarrays/solution.cc
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        unordered_map<int, int> m;
        m[0] = 1;
        int sum = 0, result = 0;
        for (int i = 0; i < nums.size(); ++i) {
            sum += nums[i] & 1;
            result += m[sum - k];
            ++m[sum];
        }
        return result;
    }
};
